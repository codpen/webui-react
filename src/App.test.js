import React from 'react';
import {shallow} from 'enzyme/build';
import App from './App';
import {findByTestAtrr, testStore} from "../Utils";
import {Provider} from "react-redux";
import toJson from 'enzyme-to-json';

const setUp = (initialState = {}) => {
    const store = testStore(initialState);
    const wrapper = shallow(
        <Provider store={store}>
            <App/>
        </Provider>).childAt(0).dive();
    return wrapper;
}

describe('App Component', function () {
    let wrapper;
    beforeEach(() => {
        const initialState = {
            status: {
                isConnected: false,
                jobs: {}
            },
            config: {
                providers: [],
                configDump: {}
            },
            remote: {
                configs: {},
                remotes: [],
                files: {},
                hasError: false
            },
            explorer: {
                backStacks: {},
                currentPaths: {},
                visibilityFilters: {},
                gridMode: {}
            },
            providerStatus: {
                about: {}
            }
        };
        wrapper = setUp(initialState);
    });

    it('should render without errors', function () {
        const component = findByTestAtrr(wrapper, 'appComponent');
        expect(component).toHaveLength(1);

    });

    it('should match snapshot', function () {
        expect(toJson(wrapper)).toMatchSnapshot()
    });
});