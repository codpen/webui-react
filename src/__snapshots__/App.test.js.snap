// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match snapshot 1`] = `
<div
  data-test="appComponent"
>
  <ErrorBoundary>
    <ToastContainer
      autoClose={5000}
      closeButton={[Function]}
      closeOnClick={true}
      draggable={true}
      draggablePercent={80}
      hideProgressBar={false}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      position="top-right"
      role="alert"
      rtl={false}
      transition={[Function]}
    />
    <HashRouter>
      <Suspense
        fallback={
          <div
            className="animated fadeIn pt-3 text-center"
          >
            Loading...
          </div>
        }
      >
        <Switch>
          <Route
            exact={true}
            name="Login Page"
            path="/login"
            render={[Function]}
          />
          <Route
            exact={true}
            name="Register Page"
            path="/register"
            render={[Function]}
          />
          <Route
            exact={true}
            name="Page 404"
            path="/404"
            render={[Function]}
          />
          <Route
            exact={true}
            name="Page 500"
            path="/500"
            render={[Function]}
          />
          <Route
            name="Home"
            path="/"
            render={[Function]}
          />
        </Switch>
      </Suspense>
    </HashRouter>
  </ErrorBoundary>
</div>
`;
